### ISSUE TITLE
qt6's lrelease and friends look in the wrong place for binaries

### ISSUE THREAD

**Issue by peterhoeg** [NixOS/nixpkgs#202293]
### Describe the bug

Running `lrelease some_file.pro` breaks with qt6 as it's paths to various binaries are wrong while working fine with qt5.

Take the following PR as an example which adds a qt6 build of the gpxsee application: https://github.com/NixOS/nixpkgs/pull/202131

In the pre-patch

In the `preConfigure` we do this in order to get the translation files generated:

``` sh
lrelease gpxsee.pro
```

And while that is fine with qt5, using qt6, it errors out with:

``` sh
configuring                                      
sh: /nix/store/g9ajh1yq2ds1hrinjn07ihm6bvvyg2mz-qtbase-6.4.0/libexec/lrelease-pro: not found
```

This is because `lrelease-pro` is located in `qttools.dev` and not `qtbase.out`

Changing `preConfigure` to this:

``` sh
    ${qttools.dev}/libexec/lrelease-pro gpxsee.pro
```

Then just gives us this:

``` sh
configuring
sh: /nix/store/g9ajh1yq2ds1hrinjn07ihm6bvvyg2mz-qtbase-6.4.0/libexec/lprodump: not found
```

So basically our builder is looking in the wrong place.

Cc: @nickcao @nrdxp

**Comment by NickCao** [comment]
Would you like to try the following patch?
```diff
diff --git a/pkgs/development/libraries/qt-6/modules/qttools.nix b/pkgs/development/libraries/qt-6/modules/qttools.nix
index b1cf5dd2c14..ec585df3ea4 100644
--- a/pkgs/development/libraries/qt-6/modules/qttools.nix
+++ b/pkgs/development/libraries/qt-6/modules/qttools.nix
@@ -8,4 +8,8 @@
 qtModule {
   pname = "qttools";
   qtInputs = [ qtbase qtdeclarative ];
+  postPatch = ''
+    substituteInPlace src/linguist/shared/runqttool.cpp \
+      --replace "QLibraryInfo::LibraryExecutablesPath" "$dev/libexec"
+  '';
 }
```

**Comment by nrdxp** [comment]
> Would you like to try the following patch?

gives:
```
  /build/qttools-everywhere-src-6.4.1/src/linguist/shared/runqttool.cpp:85:98: error: 'libexec' was not declared in this scope
     85 |     runQtTool(toolName, arguments, /nix/store/2sr79lcn75bjc6m88jlhl9m3j8hg5dy0-qttools-6.4.1-dev/libexec);
        |                                                                                                  ^~~~~~~
```

**Comment by NickCao** [comment]
Oops missed the enclosing quotation marks.

**Comment by NickCao** [comment]
Nailed it: https://gist.github.com/NickCao/73d817890229f24986ae030045be4351

**Comment by NickCao** [comment]
And the translations do work.

**Comment by peterhoeg** [comment]
Indeed, thanks a ton @NickCao !
